src/service/git/github-api.ts

```diff
@@ -128,10 +128,10 @@ export class GithubAPIService {
     if (err instanceof RequestError) {
       switch (err.status) {
         case 401:
-          reason = "Failed to authenticate with provided token, please use -token argument to provide a new one. You can also check your GITHUB_TOKEN environment variable and check whether the provided token is still valid.";
+          reason = "Failed to authenticate with provided token, please use --token argument to provide a new one. You can also check your GITHUB_TOKEN environment variable and check whether the provided token is still valid.";
           break;
         case 404:
-          reason = "Failed to fetch GitHub URL, please check if the URL used in -url argument is valid and if the token you are using have permissions to access it.";
+          reason = "Failed to fetch GitHub resource, please check if resource you requested does exits, the URL used in -u argument is valid, and if the token you are using have permissions to access it.";
```

Comment:
If the forked repository is not existing, we should display a warning and take the base repository instead directly. Is that what is happening ?

Reply:
Yeah you can find it here - https://github.com/kiegroup/github-action-build-chain/blob/main/src/service/checkout/checkout-service.ts#L179

Reply:
I can see there that if the fork does not exist on the owner repository, it raises an error: https://github.com/kiegroup/github-action-build-chain/blob/main/src/service/checkout/checkout-service.ts#L155-L158
Am I right ?

Reply:
It will only raise an error if it was not a `NotFoundError` as thrown [here](https://github.com/kiegroup/github-action-build-chain/blob/main/src/service/git/github-api.ts#L113)

Now if it was not a `NotFoundError` then it is an octokit error which implies a 4xx error on github api. If it is a 401/403 we don't have access to the target repository in which case the subsequent call to take the base repository will anyways fail. Similarly if it is 404 then we have put in an incorrect target repository which will again cause failures ahead.

Hence we are raising an error only if the error is an octokit error. This is to save ourselves from making useless github api calls ahead which will impact the github api rate limit